### PR TITLE
fix: wrong context flavor narrowing

### PR DESCRIPTION
**Pull request by KnorpelSenf**
Attempt to fix usage with lazy sessions

### CODE REVIEW

File: src/conversation.ts [hunk 1]
```diff
@@ -57,11 +57,12 @@ export type ConversationFlavor<C extends Context | undefined = undefined> =
...
-                & C
-                // deno-lint-ignore no-explicit-any
-                & (C extends { session: PromiseLike<any> }
-                    ? LazySessionFlavor<ConversationSessionData>
-                    : SessionFlavor<ConversationSessionData>)
+                & Omit<C, "session">
+                & (C extends LazySessionFlavor<infer V>
+                    ? LazySessionFlavor<ConversationSessionData & V>
+                    : C extends SessionFlavor<infer V>
+                        ? SessionFlavor<ConversationSessionData & V>
```
Comment (Loskir) [inline src/conversation.ts:64]: I don't think you need this operation with regular sessions

Comment (KnorpelSenf) [reply src/conversation.ts:64]: How else would you install the correct session types again?

File: src/conversation.ts [hunk 2]
```diff
@@ -53,11 +53,20 @@ export type ConversationFn<C extends Context> = (
...
     & (
-        | SessionFlavor<ConversationSessionData>
-        | LazySessionFlavor<ConversationSessionData>
+        C extends Context ? 
+                & Omit<C, "session">
+                & (C extends LazySessionFlavor<infer V>
+                    ? LazySessionFlavor<ConversationSessionData & V>
+                    : C extends SessionFlavor<infer V>
+                        ? SessionFlavor<ConversationSessionData & V>
+                    : never)
```
Comment (Loskir) [inline src/conversation.ts:65]:
```suggestion
        C extends Context ?
                C extends LazySessionFlavor<infer V>
                    ? Omit<C, "session"> & LazySessionFlavor<ConversationSessionData & V>
                    : С & SessionFlavor<ConversationSessionData>
```

**Comment by KnorpelSenf** [comment]
Sorry @Loskir, it's fixed now

**Review by Loskir — APPROVED** [review]
Thanks!

**Pull request merged**